src/OpenFeatureSDK/OpenFeatureClient.cs

```diff
@@ -17,11 +18,32 @@ namespace OpenFeatureSDK
     public sealed class FeatureClient : IFeatureClient
     {
         private readonly ClientMetadata _metadata;
-        private readonly FeatureProvider _featureProvider;
-        private readonly List<Hook> _hooks = new List<Hook>();
+        private readonly ConcurrentStack<Hook> _hooks = new ConcurrentStack<Hook>();
         private readonly ILogger _logger;
         private EvaluationContext _evaluationContext;
 
+        /// <summary>

```

Comment:
This is a little cumbersome. I am not sure if it is more or less cumbersome than the alternative.
The alternative would be to convert all the methods from expression bodies to normal function bodies, then alias the provider and method, then pass those to evaluate.